apps/frontend/src/components/ItemModal.tsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import {
+  HStack,
+  VStack,
+  Text,
+  Card,
+  CardBody,
+  Stack,
+  AspectRatio,
+  Image,
+  Divider,
+  Tag,
+  Wrap,
+  WrapItem,
+  Button,
+  Link,
+} from '@chakra-ui/react';
+
+import GeneralModal from './GeneralModal';
+import { ExternalLinkIcon } from '@chakra-ui/icons';
+
+
+type ItemModalHeaderProps = {
+    name: string;
+}
+
+function ItemModalHeader({ name }: ItemModalHeaderProps) {
+  return (
+    <HStack w={'100%'} justify={'center'}>
+      <Text fontSize={'xl'}>{name}</Text>
+    </HStack>
+  );
+}
+
+
+type ItemModalBodyProps = {
+    id: number;
+    name: string;
+    description: string;
+    imageUrl: string;
+    url: string
+    tags: string[];
+}
+
+function ItemModalBody({ id, name, description, imageUrl, url, tags }: ItemModalBodyProps) {
+  return (
+    <VStack>
+      <Card>
+        <CardBody>
+          <Stack spacing={4}>
+
+            <AspectRatio ratio={ 16 / 9 }>
+              <Image src={imageUrl || 'https://via.placeholder.com/150'} objectFit={'cover'} borderRadius={5}/>
+            </AspectRatio>
+
+            <Text>{description}</Text>
+
+            <Divider />
+
+            <Link href={url} isExternal> {name} <ExternalLinkIcon mx="2px" /> </Link>
+
+            <Divider />
```

Comment:
The header should contain this linking function, and more Item fields should appear here (price, company, isForSale/Trade)

Reply:
Noted for later.